connect.js

```diff
@@ -178,8 +179,17 @@ const wrap = (Wrapped, module, epics, logger, options = {}) => {
     logger.log('connect-lifecycle', `mapState for <${Wrapped.name}>, resources =`, resources);
 
     const resourceData = {};
+
     for (const r of resources) {
-      resourceData[r.name] = Object.freeze(_.get(state, [`${r.stateKey()}`], null));
+      let initState = _.get(state, r.stateKey());
+
+      if (!initState) {
+        initState = (r instanceof OkapiResource)
```

Comment:
Could you please explain the branching logic here? I see that if a resource is `OkapiResource` then it will be set with `initialResourceState` and thus the problem with `null` value is resolved. I wonder what exactly happens if the resource is not an okapi one.

Reply:
In the case of a non okapi resource (currently only `LocalResource`) the default `initialValue` defined on the given manifest will be used. 

For example:

https://github.com/folio-org/ui-users/blob/master/src/routes/UserSearchContainer.js#L31-L33

Reply:
Sounds clearer, thanks. Now I wonder how those initial values got assigned before your change here.

Reply:
@VictorSoroka1 it's a good question! The values do get assigned eventually via reducer here:

https://github.com/folio-org/stripes-connect/blob/master/LocalResource.js#L39

which is executed for each resource  in `connect`:

https://github.com/folio-org/stripes-connect/blob/master/connect.js#L106

unfortunately the `Wrapper` constructor:

https://github.com/folio-org/stripes-connect/blob/master/connect.js#L68

executes after `Wrapper.mapState`:

https://github.com/folio-org/stripes-connect/blob/master/connect.js#L68

which causes the initial values to be always empty during the first `render`.

Reply:
Thank you for a thorough explanation @mkuklis.